deltachat-ffi/src/lib.rs

```diff
@@ -1957,9 +1968,8 @@ pub unsafe extern "C" fn dc_msg_get_timestamp(msg: *mut dc_msg_t) -> i64 {
         eprintln!("ignoring careless call to dc_msg_get_received_timestamp()");
         return 0;
     }
-
-    let msg = &*msg;
-    message::dc_msg_get_timestamp(msg)
+    let ffi_msg = &*msg;
+    message::dc_msg_get_timestamp(&ffi_msg.message)
```

Comment:
you can implement `Deref<Target=Message>`  for `MessageWrapper`, this way you can write `message::dc_msg_get_timestamp(&ffi_msg)` etc

Reply:
This seemed like a reasonable idea so I started looking into it and found this in the docs:
"Deref should **only** be implemented for smart pointers"

It's a bit disputable whether this qualifies I think, so I'll leave it.